File: algorithm/src/main/java/baekjoon/p02754/Main.java
package baekjoon.p02754;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.next();
        System.out.println(solve(input));
    }

    private static double solve(String input) {
        if (input.equals("F")) {
            return 0;
        }

        double ret = 1;
        ret += ('D' - input.charAt(0));
        ret += getPoint(input.charAt(1));

        return ret;
    }

    private static double getPoint(char ch) {
        switch (ch) {
            case '+':
                return 0.3;
            case '-':
                return -0.3;
            default:
                return 0;
        }
    }

}

File: algorithm/src/main/java/baekjoon/p02754/Main.kt
package baekjoon.p02754

fun main(args: Array<String>) {
    val input = readLine().toString()
    println(solve(input))
}

private fun solve(input: String): Double {
    if (input == "F") {
        return 0.0
    }

    var ret = 1.0
    ret += ('D' - input[0]).toDouble()
    ret += when (input[1]) {
        '+' -> 0.3
        '-' -> -0.3
        else -> 0.0
    }

    return ret
}